pkg/ingester/ingester.go

```diff
@@ -354,6 +358,43 @@ func (i *Ingester) QuerySample(req *logproto.SampleQueryRequest, queryServer log
 	return sendSampleBatches(queryServer.Context(), heapItr, queryServer)
 }
 
+func (i *Ingester) GetChunkIDs(ctx context.Context, req *logproto.GetChunkIDsRequest) (*logproto.GetChunkIDsResponse, error) {
+	orgID, err := user.ExtractOrgID(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// when using boltdb-shipper, limit the request start time to the start of current schema config otherwise
+	// if query covers non boltdb-shipper stores too then we would un-necessarily be querying them as well which would anyways be done by queriers.
+	reqStart := req.Start
+	if i.store.ActivePeriodConfig().IndexType == shipper.BoltDBShipperType {
+		reqStart = adjustQueryStartTime(time.Since(i.store.ActivePeriodConfig().From.Time.Time()), reqStart, time.Now())
+	}
+
+	// parse the request
+	start, end := listutil.RoundToMilliseconds(reqStart, req.End)
+	matchers, err := logql.ParseMatchers(req.Matchers)
+	if err != nil {
+		return nil, err
+	}
+
+	// get chunk references
+	chunksGroups, _, err := i.store.GetChunkRefs(ctx, orgID, start, end, matchers...)
+	if err != nil {
+		return nil, err
+	}
+
+	// build the response
+	resp := logproto.GetChunkIDsResponse{ChunkIDs: []string{}}
```

Comment:
could pre-allocate this with make() to save a few allocations.

Reply:
We are iterating over `chunks` nested inside `chunksGroups` and I do not see an easier way to estimate the total number of chunks. Am I missing something?

Reply:
nope, it was late and I didn't look at this close enough. Looks good!